client/state/data-layer/wpcom/gravatar-upload/index.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * Internal dependencies
+ */
+import {
+	GRAVATAR_UPLOAD_RECEIVE,
+	GRAVATAR_UPLOAD_REQUEST,
+	GRAVATAR_UPLOAD_REQUEST_SUCCESS,
+	GRAVATAR_UPLOAD_REQUEST_FAILURE,
+} from 'state/action-types';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import {
+	bumpStat,
+	composeAnalytics,
+	recordTracksEvent,
+	withAnalytics,
+} from 'state/analytics/actions';
+
+function uploadGravatar( { dispatch }, action, next ) {
+	const { email, file } = action;
+	dispatch( http( {
+		method: 'POST',
+		path: '/gravatar-upload',
+		body: {},
+		apiNamespace: 'wpcom/v2',
+		formData: [
+			[ 'account', email ],
+			[ 'filedata', file ],
+		],
+	}, action ) );
+
+	next( action );
```

Comment:
without this `next()` available the original `_REQUEST` action will automatically pass through to the reducers. if we have to dispatch different actions (for example, suppose we had to also grab user details and dispatch `USER_REQUEST`) then we just "start over" with `store.dispatch`.

on the other hand, if we need to pass through an action which could cause a cycle/loop then we must wrap it in `local` so that it bypasses the data layer

```js
store.dispatch( local( action ) )
```

only in rare cases should we need to send off a local action